libs/execution_context/ebpf_maps.c

```diff
@@ -1424,6 +1424,15 @@ _update_hash_map_entry_with_handle(
         goto Done;
     }
 
+    // Store the new object ID as the value.
+    result =
+        ebpf_hash_table_update((ebpf_hash_table_t*)map->data, key, (uint8_t*)&value_object->id, hash_table_operation);
+    if (result != EBPF_SUCCESS) {
+        EBPF_LOG_MESSAGE_NTSTATUS(
+            EBPF_TRACELOG_LEVEL_ERROR, EBPF_TRACELOG_KEYWORD_MAP, "Hash table update failed.", result);
+        goto Done;
+    }
+
     // Release the reference on the old ID stored here, if any.
     ebpf_id_t old_id = (old_value) ? *(ebpf_id_t*)old_value : 0;
```

Comment:
Offhand this change doesn't look safe but I can't tell for sure.  That is, old_value points into the space for the table.  If the update in 142 overwrites that memory with the new value in old_id will be set here to the new id not the old id.
So I think line 1437 needs to come before line 1428

Reply:
I see. The best way to address will be in the following order:
1. Finding the existing entry in the table and then validate. 
2. Store the old ID  
3. Store the new object ID (which will replace the old ID)
4. Check to see if old ID exists and release 
5. Acquire on the ref count.